commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls

```diff
@@ -0,0 +1,147 @@
+/*
+ * @description Example of unit test for CreateOrder extension point. This test is mocking out default implementation of CreateOrder Extension so
+ * the test is isolated from the dependencies on other services that default implementation might have.
+ */
+@isTest
+public class CreateOrderSampleUnitTest {
+    
+    @isTest
+    public static void shouldRoundUpCostOfProduct() {
+        CartExtension.Cart cart = arrangeCart(new Decimal[]{1.5});
+        CreateOrderSample createOrder = new CreateOrderSampleMock();
+        // Act
+        Test.startTest();
+        CartExtension.CreateOrderResponse response = createOrder.createOrder(new CartExtension.CreateOrderRequest(cart, '123'));
+        Test.stopTest();
+        // Assert
+        Order order = response.getOrderGraph().getOrder();
+        Assert.areEqual('Total rounded up cost of products: 2.0', order.Description);
+    }
+
+    @isTest
+    public static void shouldRoundUpCostOfSeveralProducts() {
+        CartExtension.Cart cart = arrangeCart(new Decimal[]{1.5, 2.6});
+        CreateOrderSample createOrder = new CreateOrderSampleMock();
+        // Act
+        Test.startTest();
+        CartExtension.CreateOrderResponse response = createOrder.createOrder(new CartExtension.CreateOrderRequest(cart, '123'));
+        Test.stopTest();
+        // Assert
+        Order order = response.getOrderGraph().getOrder();
+        Assert.areEqual('Total rounded up cost of products: ' + (2.0 + 3.0), order.Description);
+    }
+    
+    /*
+     * Mock of CreateOrderSample class. It provides simplified implementation of cart to order mapping. 
+     */    
+    public class CreateOrderSampleMock extends CreateOrderSample {
+        
+        public override CartExtension.CreateOrderResponse callDefault(CartExtension.CreateOrderRequest request) {
+            CartExtension.Cart cart = request.getCart();
+            CartExtension.OrderGraph orderGraph = new CartExtension.OrderGraph();
+            Order order = orderGraph.getOrder();
+            System.assert(order.Id != null);
+    
+            order.AccountId = cart.getAccountId();
+            order.OwnerId = cart.getOwnerId();
+            order.SalesStoreId = cart.getWebStoreId();
+            // cart does not support PoNumber yet
+            //order.PoNumber = cart.getPoNumber();
```

Comment:
Do we want to keep the commented out code?

Reply:
As we going to use HSOI create order, we need to ensure those fields in the Order entities are available in summary entities.

Reply:
to be on the safe side I'll remove fields that are not in the OrderSummary entity. We should start spiking about HSOI mapping. We may find many non supported fields.